### ISSUE TITLE
Creating a "Resource Group" with the same doesn't give an error

### ISSUE THREAD

**Issue by hugo-paredes** [terraform-providers/terraform-provider-azurerm#5549]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and AzureRM Provider) Version

`terraform v0.12.20`

### Affected Resource(s)

* `azurerm_resource_group`

### Terraform Configuration Files

```hcl
resource "azurerm_resource_group" "this" {
  name     = "some-name-already-exists"
  location = "West US"
}
```

### Expected Behavior

Because the `resource group` resource already exists, I would expect it would give an error or throw an exception saying that a `resource group` with that name already exists.

### Actual Behavior

Instead, `terraform` ignored the fact that `resource group` already existed, and created it and all other resources in the script.

When I ran the `terraform destroy` command, it deleted every single resource part of the `tfstate`, including the `resource group` that it didn't created. As a result, all other resources that were part of that `resource group` and were created elsewhere were, those resources were also deleted.

### Steps to Reproduce

1. `terraform apply`
2. `terraform destroy`

**Comment by tombuildsstuff** [comment]
hi @hugo-paredes 

Thanks for opening this issue.

In Azure the majority of API calls are now Upserts rather than separate Create/Update calls - and as such it's possible for resources (such as in this case the Resource Group) to be "adopted" into Stateful tools such as Terraform.

Unfortunately this bug is an oversight which has existed since the early days of the Provider - from our side we identified this a while back and (since introducing this would be a breaking behavioural change) plan to fix it as a part of the upcoming major/2.0 release (more details can be found in #2807). The fix for this has already been merged and can be enabled via the Feature Toggle, as detailed in #2807 which first checks for the presence of an existing resource with the same identifier when trying to create a new resource.

As this will be fixed by #2807 I'm going to close this issue for the moment - but would you mind subscribing to that issue for updates? In the interim you should be able to feature toggle this behaviour on in 1.x versions of the Provider by setting the Environment Variable `ARM_PROVIDER_STRICT` to `true`.

Thanks!

**Comment by hugo-paredes** [comment]
Can I at least recommend some sort of acknowledgment in the (documentation0[https://www.terraform.io/docs/providers/azurerm/r/resource_group.html] page? There's nothing there indicating that this can happen and it can be very problematic.

Thanks.
Hugo
